core/src/main/java/org/springframework/security/extensions/saml2/config/SAMLConfigurer.java

```diff
@@ -124,6 +125,12 @@ public void init(HttpSecurity http) {
 		cachingMetadataManager = cachingMetadataManager();
 		webSSOProfile = webSSOProfile();
 		singleLogoutProfile = singleLogoutProfile();
+
+		if (webSSOProfileConsumer == null) {
```

Comment:
OK so I'm not actually super happy with having to do this but I wanted to open the PR to discuss.

The main issue is that currently there's the `webSSOProfileConsumer` method that sets the `webSSOProfileConsumer` that's public. As far as I can tell, the main purpose of that setter is to allow the `webSSOProfileConsumer` to be mocked out in tests.

Having that method be public is problematic though, because end users can't actually construct their own `WebSSOProfileConsumerImpl`, as they don't have access to the `SAMLProcessor` and `MetadataManager` that need to be injected into it for it to work. It's fine when mocking because no real object gets constructed, but having that method be public leads to incorrect uses, like here: https://github.com/spinnaker/gate/pull/550

I'd like to remove the `webSSOProfileConsumer`, or at the very least make it package-private or something so it can still be used in tests. It's possible I'm missing something with how Spring DI works, but is there a better way to inject a mock here for tests? I don't know if its possible to feed this Configurer class thru spring's DI so the actual `WebSSOProfileConsumer` implementation can be chosen at runtime? (I'm not actually super well-versed in Spring, so I'll defer to an expert here)

Reply:
This seems like two separate issues. You are first interested in adding the ability to have max authentication age and second want to remove the setter. Does that sound correct? If so, let's create another issue for the possibility to remove webSSOProfileConsumer

Reply:
Oh yeah, sorry, I didn't mean to imply I wanted to do that as part of this PR :) I just wanted to note it here to see if you had any thoughts